src/result_processor.c

```diff
@@ -891,6 +892,8 @@ static int rpSafeLoaderNext_Accumulate(ResultProcessor *rp, SearchResult *res) {
   // Done loading. Unlock Redis
   RedisModule_ThreadSafeContextUnlock(sctx->redisCtx);
 
+  rp->totalGILTime += clock() - start;
```

Comment:
Is this used for statistics as well or only for profiling?
If only for profiling, consider calling `clock()` conditionally to avoid performance overhead